watertap/costing/watertap_costing_package.py

```diff
@@ -0,0 +1,501 @@
+                initialize=535,
+                doc='Pressure exchanger cost',
+                units=self.base_currency/(pyo.units.meter**3/pyo.units.hours))
+        self.pressure_exchanger_cost = pyo.Var(
+                initialize=535,
+                doc='Pressure exchanger cost',
+                units=self.base_currency/(pyo.units.meter**3/pyo.units.hours))
+        self.energy_recovery_device_linear_coefficient = pyo.Var(
+                initialize=3134.7,
+                doc='Energy recovery device linear coefficient',
+                units=self.base_currency)
+        self.energy_recovery_device_exponent = pyo.Var(
+                initialize=0.58,
+                doc='Energy recovery device exponent',
+                units=pyo.units.dimensionless)
+        self.separator_unit_cost = pyo.Var(
+                initialize=361,
+                doc='Separator cost',
+                units=self.base_currency/(pyo.units.liters/pyo.units.second))
+        self.mixer_unit_cost = pyo.Var(
+                initialize=361,
+                doc='Mixer cost',
+                units=self.base_currency/(pyo.units.liters/pyo.units.second))
+        self.naocl_mixer_unit_cost = pyo.Var(
+                initialize=5.08,
+                doc='NaOCl mixer cost',
+                units=self.base_currency/(pyo.units.m**3/pyo.units.day))
+        self.caoh2_mixer_unit_cost = pyo.Var(
+                initialize=792.8*2.20462,
+                doc='Ca(OH)2 mixer cost',
+                units=self.base_currency/(pyo.units.kg/pyo.units.day))
+
+        self.electricity_base_cost = pyo.Param(
+                mutable=True,
+                initialize=0.07,
+                doc='Electricity cost',
+                units=self.base_currency/pyo.units.kWh)
+        self.naocl_cost = pyo.Param(
+                initialize=0.23,
+                doc='NaOCl cost',
+                units=self.base_currency/pyo.units.kg)
+        self.naocl_purity = pyo.Param(
+                mutable=True,
+                initialize=0.15,
+                doc='NaOCl purity',
+                units=pyo.units.dimensionless)
+        self.caoh2_cost = pyo.Param(
+                mutable=True,
+                initialize=0.12,
+                doc='CaOH2 cost',
+                units=self.base_currency/pyo.units.kg)
+        self.caoh2_purity = pyo.Param(
+                mutable=True,
+                initialize=1,
+                doc='CaOH2 purity',
+                units=pyo.units.dimensionless)
+
+        # fix the parameters
+        for var in self.component_objects(pyo.Var):
+            var.fix()
+
+        # Define standard material flows and costs
+        self.defined_flows["electricity"] = self.electricity_base_cost
+        self.defined_flows["NaOCl"] = 74.44e-3 * (pyo.units.kg/pyo.units.mol) * (self.naocl_cost/self.naocl_purity)
+        self.defined_flows["CaOH2"] = 74.093e-3 * (pyo.units.kg/pyo.units.mol) * (self.caoh2_cost/self.caoh2_purity)

```

Comment:
Should these values (74.44e-3 and 74.093e-3) be Params? These are the molecular weights of each chemical.

Reply:
Maybe? My main thought on hard-coding them is that unlike `electricity_base_cost`, I can't imagine needing a reason for the user to set these.

Reply:
Out of interest, why do you even need the mole to mass conversion? Costing supports both mass and mole based flows, so it would make more sense to have the cost and flow use the same units and not have to worry about conversion.

Reply:
I doubt it is strictly necessary -- I'm just replicating what was in the financials module.